From 4b747c468fb27a662d544659bf7ee89c9abb48dd Mon Sep 17 00:00:00 2001
From: Rajat Soni <quic_rajson@quicinc.com>
Date: Fri, 11 Nov 2022 14:26:27 +0530
Subject: [PATCH] hostapd: enable EHT320 mode in 5 GHz

Currently, EHT320 mode is disabled in case non 6 GHz channels. However, we have
support to handle EHT320 mode with punctured 80 MHz sub-channel in 5 GHz.

Hence, enable EHT320 mode in non 6 GHz channels.

Signed-off-by: Rajat Soni <quic_rajson@quicinc.com>
Signed-off-by: Aditya Kumar Singh <quic_adisi@quicinc.com>
---
 src/common/hw_features_common.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/hw_features_common.c b/src/common/hw_features_common.c
index 3e43d82..186b4b5 100644
--- a/src/common/hw_features_common.c
+++ b/src/common/hw_features_common.c
@@ -685,9 +685,9 @@ int hostapd_set_freq_params(struct hostapd_freq_params *data,
 		break;
 	case CONF_OPER_CHWIDTH_320MHZ:
 		data->bandwidth = 320;
-		if (!data->eht_enabled || !is_6ghz_freq(freq)) {
+		if (!data->eht_enabled) {
 			wpa_printf(MSG_ERROR,
-				   "320 MHz: EHT not enabled or not a 6 GHz channel");
+				   "320 MHz: EHT not enabled");
 			return -1;
 		}
 		if (center_segment1) {
-- 
2.17.1

